x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -455,6 +478,19 @@ private void prewarmNext(final Executor executor, final BlockingQueue<Tuple<Acti
         }
     }
 
+    private void addRecoveryFileDetails() {
```

Comment:
Inline this into `prewarmCache`? Especially as it's duplicating the iteration logic down below.